WcaOnRails/app/controllers/competitions_controller.rb

```diff
@@ -69,6 +62,15 @@ def index
       @competitions = @competitions.select { |competition| competition.contains?(params[:search]) }
     end
 
+    unless params[:event_ids].empty?
+      @competitions = @competitions.select do |competition|
+        (competition.eventSpecs.split & params[:event_ids]).any?
```

Comment:
I'm not wild about the implementation details of `eventSpecs` leaking out of the Competition model here. Would something like this work?

```ruby
  required_events = params[:event_ids].map { |event_id| Event.find(event_id) }
  unless params[:event_ids].empty?
    @competitions = @competitions.select do |competition|
      (competition.events & required_events).any?
```